src/strings.cpp

```diff
@@ -1228,7 +1229,7 @@ static char *FormatString(char *buff, const char *str_arg, StringParameters *arg
 
 			case SCC_VELOCITY: { // {VELOCITY}
 				assert(_settings_game.locale.units_velocity < lengthof(_units_velocity));
-				int64 args_array[] = {ConvertKmhishSpeedToDisplaySpeed(args->GetInt64(SCC_VELOCITY))};
+				int64 args_array[] = {ConvertKmhishSpeedToDisplaySpeed(args->GetInt64(SCC_VELOCITY)), 1 /* decimal places */};
```

Comment:
While this appears to be working, it is not technically correct, in the sense that for the old units you now have this extra string parameter, which will get ignored in this case, but might cause problems in the future.

You should better take care of this now, instead of hoping it won't fall apart.

Options could be:
 * branching the code so you have 1 or 2 parameters, depending on the unit used
 * converting the other units from `{COMMA}` to `{DECIMAL}` as well, with second parameter of 0 (no idea how that gets displayed)

Reply:
> You should better take care of this now, instead of hoping it won't fall apart.

To be clear, I wasn't "hoping it won't fall apart". I knew that commit was incomplete, but I didn't see _how_ it was incomplete, so I pushed it and indicated which part of it I suspected to be incorrect. I never expected the PR to be merged in that state.

> branching the code so you have 1 or 2 parameters, depending on the unit used

I took a shot at this, using another StringParameters constructor so I can pass the array length explicitly. Seems good enough for now, and can be upgraded to std::vector in the future if needed.

> converting the other units from `{COMMA}` to `{DECIMAL}` as well, with second parameter of 0 (no idea how that gets displayed)

`{DECIMAL}` with a second parameter of 0 is synonymous with `{COMMA}`, if I'm reading the code right:

https://github.com/OpenTTD/OpenTTD/blob/a56bf35409919212da9907fcdd9bc33929b4ac86/src/strings.cpp#L1045-L1053
https://github.com/OpenTTD/OpenTTD/blob/a56bf35409919212da9907fcdd9bc33929b4ac86/src/strings.cpp#L343

Let me know if you prefer this approach. (How are translations handled, though? Should I change `{COMMA}` to `{DECIMAL}` in all translations, or leave that to translators?)